### ISSUE TITLE
Show all queries that are being fired on a pivot table

### ISSUE THREAD

**Issue by paoliniluis** [metabase/metabase#25037]
**Describe the bug**
Right now pivot tables fire several queries to the DW, but we only show one (the main one). It would be a good idea to expose all the queries so users don't see an asymmetry between their clusters and Metabase

**Logs**
```
2022-08-26 17:52:01.627 UTC [29] LOG:  execute <unnamed>/C_1: -- Metabase:: userID: 1 queryType: MBQL queryHash: 7329b8743f478a183f6ca1a52709ece03952ed684f0f8eccac88c33b7a33763a
	SELECT "people__via__user_id"."source" AS "people__via__user_id__source", sum("public"."orders"."subtotal") AS "sum", sum("public"."orders"."total") AS "sum_2" FROM "public"."orders" LEFT JOIN "public"."people" "people__via__user_id" ON "public"."orders"."user_id" = "people__via__user_id"."id" GROUP BY "people__via__user_id"."source" ORDER BY "people__via__user_id"."source" ASC
2022-08-26 17:52:06.661 UTC [29] LOG:  execute <unnamed>/C_3: -- Metabase:: userID: 1 queryType: MBQL queryHash: cefc0421e45490880b0fd6de15b418d9639f62a487c3bd7906e77dcda839c0d2
	SELECT "source"."people__via__user_id__source" AS "people__via__user_id__source", "source"."pivot-grouping" AS "pivot-grouping", sum("source"."subtotal") AS "sum", sum("source"."total") AS "sum_2" FROM (SELECT "public"."orders"."id" AS "id", "public"."orders"."user_id" AS "user_id", "public"."orders"."product_id" AS "product_id", "public"."orders"."subtotal" AS "subtotal", "public"."orders"."tax" AS "tax", "public"."orders"."total" AS "total", "public"."orders"."discount" AS "discount", "public"."orders"."created_at" AS "created_at", "public"."orders"."quantity" AS "quantity", abs(0) AS "pivot-grouping", "people__via__user_id"."source" AS "people__via__user_id__source", "people__via__user_id"."id" AS "people__via__user_id__id" FROM "public"."orders" LEFT JOIN "public"."people" "people__via__user_id" ON "public"."orders"."user_id" = "people__via__user_id"."id") "source" GROUP BY "source"."people__via__user_id__source", "source"."pivot-grouping" ORDER BY "source"."people__via__user_id__source" ASC, "source"."pivot-grouping" ASC
2022-08-26 17:52:06.714 UTC [29] LOG:  execute <unnamed>/C_4: -- Metabase:: userID: 1 queryType: MBQL queryHash: c9a8f2c71b43431d986199882a5b03de235769b4385974b7c8fb97371704581d
	SELECT "source"."pivot-grouping" AS "pivot-grouping", sum("source"."subtotal") AS "sum", sum("source"."total") AS "sum_2" FROM (SELECT "public"."orders"."id" AS "id", "public"."orders"."user_id" AS "user_id", "public"."orders"."product_id" AS "product_id", "public"."orders"."subtotal" AS "subtotal", "public"."orders"."tax" AS "tax", "public"."orders"."total" AS "total", "public"."orders"."discount" AS "discount", "public"."orders"."created_at" AS "created_at", "public"."orders"."quantity" AS "quantity", abs(1) AS "pivot-grouping" FROM "public"."orders") "source" GROUP BY "source"."pivot-grouping" ORDER BY "source"."pivot-grouping" ASC
2022-08-26 17:52:06,729 DEBUG middleware.log :: POST /api/dataset/pivot 202 [ASYNC: completed] 190.2 ms (26 DB calls) App DB connections: 0/4 Jetty threads: 5/50 (0 idle, 0 queued) (47 total active threads) Queries in flight: 0 (0 queued); postgres DB 2 connections: 0/2 (0 threads blocked)
```

**To Reproduce**
1) new question -> Orders, summarize-> sum of subtotal and sum of total group by user->source
2) see the SQL generated

on the logs, you see that there are 3 queries being fired, not just one

on bigger pivot tables there are even more

**Expected behavior**
expose all activity to the user

**Screenshots**
NA

**Information about your Metabase Installation:**
- Your browser and the version: Brave latest
- Your operating system: Pop 22.04
- Your databases: Postgres 14.2
- Metabase version: 44.1
- Metabase hosting environment: Docker
- Metabase internal database: Postgres 14.2

**Severity**
P3

**Additional context**
NA

**Comment by calherries** [comment]
What did you mean by "see the SQL generated" in the 2nd step here @paoliniluis? Where can the user see the SQL generated? Do you mean in `data.native_form.query` of the response for the POST /api/dataset/pivot request?

**Comment by paoliniluis** [comment]
@calherries, out pivot table implementation currently fires many queries to the DB, and not only the one that Metabase shows on the "SQL for this question". Until we manage to do real pivot queries (https://github.com/metabase/metabase/issues/29348 and probably https://github.com/metabase/metabase/issues/21688) we should show all the queries that MB does to the DB when it tries to build a pivot table

**Comment by paoliniluis** [comment]
To add some detail on how we build the queries here:

The queries have the following structure
cell values 
SELECT "source"."state" AS "state", "source"."source" AS "source", "source"."pivot-grouping" AS "pivot-grouping", SUM("source"."product_id") AS "sum", SUM("source"."tax") AS "sum_2" FROM (SELECT "source"."id" AS "id", "source"."user_id" AS "user_id", "source"."product_id" AS "product_id", "source"."subtotal" AS "subtotal", "source"."tax" AS "tax", "source"."total" AS "total", "source"."discount" AS "discount", "source"."created_at" AS "created_at", "source"."quantity" AS "quantity", "source"."email" AS "email", "source"."state" AS "state", "source"."source" AS "source", ABS(0) AS "pivot-grouping" FROM (select * from "metabase_cache_18480_2"."model_16_model_1_1_") AS "source") AS "source" GROUP BY "source"."state", "source"."source", "source"."pivot-grouping" ORDER BY "source"."state" ASC, "source"."source" ASC, "source"."pivot-grouping" ASC
 
2. Row totals
SELECT "source"."source" AS "source", "source"."pivot-grouping" AS "pivot-grouping", SUM("source"."product_id") AS "sum", SUM("source"."tax") AS "sum_2" FROM (SELECT "source"."id" AS "id", "source"."user_id" AS "user_id", "source"."product_id" AS "product_id", "source"."subtotal" AS "subtotal", "source"."tax" AS "tax", "source"."total" AS "total", "source"."discount" AS "discount", "source"."created_at" AS "created_at", "source"."quantity" AS "quantity", "source"."email" AS "email", "source"."state" AS "state", "source"."source" AS "source", ABS(1) AS "pivot-grouping" FROM (select * from "metabase_cache_18480_2"."model_16_model_1_1_") AS "source") AS "source" GROUP BY "source"."source", "source"."pivot-grouping" ORDER BY "source"."source" ASC, "source"."pivot-grouping" ASC
 
3. Column totals
SELECT "source"."state" AS "state", "source"."pivot-grouping" AS "pivot-grouping", SUM("source"."product_id") AS "sum", SUM("source"."tax") AS "sum_2" FROM (SELECT "source"."id" AS "id", "source"."user_id" AS "user_id", "source"."product_id" AS "product_id", "source"."subtotal" AS "subtotal", "source"."tax" AS "tax", "source"."total" AS "total", "source"."discount" AS "discount", "source"."created_at" AS "created_at", "source"."quantity" AS "quantity", "source"."email" AS "email", "source"."state" AS "state", "source"."source" AS "source", ABS(2) AS "pivot-grouping" FROM (select * from "metabase_cache_18480_2"."model_16_model_1_1_") AS "source") AS "source" GROUP BY "source"."state", "source"."pivot-grouping" ORDER BY "source"."state" ASC, "source"."pivot-grouping" ASC
 
4. Grand totals
SELECT "source"."pivot-grouping" AS "pivot-grouping", SUM("source"."product_id") AS "sum", SUM("source"."tax") AS "sum_2" FROM (SELECT "source"."id" AS "id", "source"."user_id" AS "user_id", "source"."product_id" AS "product_id", "source"."subtotal" AS "subtotal", "source"."tax" AS "tax", "source"."total" AS "total", "source"."discount" AS "discount", "source"."created_at" AS "created_at", "source"."quantity" AS "quantity", "source"."email" AS "email", "source"."state" AS "state", "source"."source" AS "source", ABS(3) AS "pivot-grouping" FROM (select * from "metabase_cache_18480_2"."model_16_model_1_1_") AS "source") AS "source" GROUP BY "source"."pivot-grouping" ORDER BY "source"."pivot-grouping" ASC
 
 
(column, row and grand totals are fired no matter if you have those enabled or not)

If you see there's a ABS(x) field which is being used as "pivot-grouping" and it's there only to get what the query is (cell or totals)

**Comment by darksciencebase** [comment]
@luizarakaki

**Comment by paoliniluis** [comment]
@luizarakaki yup

**Comment by kamilmielnik** [comment]
Reproducible in `master` at 950a275387

**Comment by camsaul** [comment]
I don't think this would be super hard to do, as Cal mentioned we know what SQL we're generating before we run anything, it would be easy enough to special case the API endpoint for getting the underlying SQL (or QP results?) here for pivot queries.
